mars/services/web/supervisor.py

```diff
@@ -86,13 +87,15 @@ def __init__(self, config):
         super().__init__()
         self._config = config
         self._web_server = None
+        self._web_address = None
 
     async def start(self):
         static_path = os.path.join(os.path.dirname(__file__), 'static')
         supervisor_addr = self.address
 
-        host = self._config.get('host') or '0.0.0.0'
-        port = self._config.get('port')
+        host = self._config.get('host', socket.gethostbyname(socket.gethostname()))
```

Comment:
This may not work when DNS is not accessible.

Reply:
Added `mars.utils.get_node_ip_address` implementation from ray